app.js

```diff
@@ -0,0 +1,149 @@
+function shipPlacementGeneration()
+{
+    createGrid(getContainer("player-grid-container"), true);
+    createGrid(getContainer("computer-grid-container"), false);
+}
+
+function createGrid(container, isUser)
+{
+    for(let i = 0; i < 10; i++)
+    {
+        for(let j = 0; j < 10; j++)
+        {
+            var item = document.createElement("div");
+            if(isUser)
+            {
+                item.id = "user";
+                item.id += " square-" + i + "-" + j;
+            }
+            else
+            {
+                item.id = "computer";
+                item.id += " square-" + i + "-" + j;
+            }
+            item.className += " square";
+            container.appendChild(item);
+        }
+    }
+}
+
+function getContainer(id)
+{
+    return document.getElementById(id);
+}
+
+function getShipRandomPosition() 
+{
+    return { 
+        x : Math.floor(Math.random() * 10),
+        y : Math.floor(Math.random() * 10),
+        isHorizontal : isHorizontal = Math.random() > 0.5 
+    }
+}
+
+function createShipCoordinates(squaresCount)
+{
+    var shipCoordinates = [];
+    
+    for(let ship = 0; ship < squaresCount.length; ship++)
+    {
+        var position = getShipRandomPosition();
+    
+        if (position.isHorizontal)
+        {
+            if(10 - squaresCount[ship] >= position.x)
+            {
+                for (let length = 0; length < squaresCount[ship]; length++)
+                {   
+                    shipCoordinates.push("ship-part-" + position.y + "-" + position.x);
```

Comment:
Why do you need this string generation?
You can just create the boolean array 10x10 and set true where the ship should be